In [1]:
import pandas as pd
import numpy as np
import statistics as stats
import matplotlib.pyplot as plt
%matplotlib inline

brady_kids = pd.DataFrame()
brady_kids['age'] = [14, 12, 11, 10, 8, 6, 8]
brady_kids.index = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Cousin Oliver']
brady_kids

age
Greg            14
Marcia          12
Peter           11
Jan             10
Bobby            8
Cindy            6
Cousin Oliver    8

In [2]:
brady_kids.describe()

age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [3]:
print(np.mean(brady_kids['age']))
print(np.median(brady_kids['age']))
print(stats.mode(brady_kids['age']))

9.857142857142858
10.0
8


In [4]:
print(np.var(brady_kids['age']))
print(np.std(brady_kids['age']))
print(np.std(brady_kids['age']) / (np.sqrt(len(brady_kids['age']))))

6.408163265306122
2.531435020952764
0.9567925036515135


Of the central tendancy measures, mean and median both provide a similar description of the data.  I might choose median since it is a whole number and a good representation of the ages.

The standard deviation provides the best description of the variance measures because it is given in a unit that makes it easier to understand how the data is spread.

In [5]:
brady_kids.loc['Cindy', 'age'] += 1
brady_kids

age
Greg            14
Marcia          12
Peter           11
Jan             10
Bobby            8
Cindy            7
Cousin Oliver    8

In [6]:
print(np.mean(brady_kids['age']))
print(np.median(brady_kids['age']))
print(stats.mode(brady_kids['age']))

10.0
10.0
8


In [7]:
print(np.var(brady_kids['age']))
print(np.std(brady_kids['age']))
print(np.std(brady_kids['age']) / (np.sqrt(len(brady_kids['age']))))

5.428571428571429
2.32992949004287
0.8806305718527109


Median and mode stayed the same.

Mean, variance, standard deviation, and standard error changed.

In [18]:
brady_kids2 = pd.DataFrame()
brady_kids2['age'] = [14, 12, 11, 10, 8, 6, 1]
brady_kids2.index = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica']
brady_kids2

age
Greg      14
Marcia    12
Peter     11
Jan       10
Bobby      8
Cindy      6
Jessica    1

In [19]:
df = pd.DataFrame()
df['fan percentage'] = [.2, .23, .17, .05]
df.index = ['TV GUide', 'Entertainment Weekly', 'Pop Culture Today', 'SciPhi Phanatic']
df

fan percentage
TV GUide                        0.20
Entertainment Weekly            0.23
Pop Culture Today               0.17
SciPhi Phanatic                 0.05

In [20]:
df.describe()

fan percentage
count        4.000000
mean         0.162500
std          0.078899
min          0.050000
25%          0.140000
50%          0.185000
75%          0.207500
max          0.230000

Based on the data, it could be estimated 16.25% of adult Americans are fans of the Brady Bunch.  However, since the surveys are of fans of entertainment magazines, this may not be a good sample of the population.